rust/operator-binary/src/restart_controller/pod.rs

```diff
@@ -104,10 +112,28 @@ async fn reconcile(pod: Arc<Pod>, ctx: Arc<Ctx>) -> Result<Action, Error> {
         })
         .transpose()?;
 
+    tracing::debug!(
+        "Proceeding with closest expiration time [{:?}]",
+        pod_expires_at
+    );
     let now = DateTime::<FixedOffset>::from(Utc::now());
+
+    // Calculate the time remaining from now until the stated expiration time by subtraction
+    // The call to `chrono::Duration::to_std()` returns an error if the resulting duration is
+    // negative -> i.e. when the certificate has expired.
+
     let time_until_pod_expires = pod_expires_at.map(|expires_at| (expires_at - now).to_std());
+
+    // Match on result of subtraction, possible cases:
+    // Some(Error<...>) -> duration was negative, cert has expired
+    // Some(Ok<Duration<>>) -> duration was positive, cert still valid
+    // None -> there were no annotations to process, pod is not in scope for this code
     match time_until_pod_expires {
         Some(Err(_has_already_expired)) => {
+            tracing::info!(
+                "Evicting pod, due to stated expiration date being reached - valid_until=[{}]",
+                pod_expires_at.unwrap()
```

Comment:
```suggestion
                pod.expires_at = %pod_expires_at.unwrap(),
                "Evicting pod, due to stated expiration date being reached",
```

Reply:
That said, not a fan of this `unwrap`.. we could restructure `let pod_expires_at` as a `let ... else` block instead to avoid it (or wrap all of this in a `match` instead of mapping over `pod_expires_at`).

Reply:
For now my goal was to change actual code as little as possible, but I am happy to restructure this of course.

Reply:
I simply removed the .unwrap(), so the logs look like this now:

> Pod still valid, rescheduling check pod.expires_at=Some(2024-02-01T03:26:35.416278954+00:00) recheck_delay=64799.599882683s

No reason not to log that information along with it I think?

Reply:
Sure, that's fine.

Happy to submit a followup or suggest a change for the restructuring later, but also wouldn't consider it blocking for this PR.

Reply:
works for me